dashboard/app.py

```diff
@@ -75,37 +68,36 @@ def disable_varnish(response):
         setup_authentication(app)
 
     app.register_blueprint(page)
-    app.register_blueprint(extra)
+
+    plugins = {}
+    for plugin in app.config['ENABLED_PLUGINS']:
+        module = importlib.import_module(f'dashboard.plugins.{plugin}')
+        app.register_blueprint(module.plugin, url_prefix=module.base_path)
+        plugins[plugin] = {'tab_name': module.tab_name}
 
     app.airflow_data_provider = AirflowDBDataProvider(app.config, app.logger, MySQLClient(app.config, app.logger))
     app.influx_data_provider = InfluxDBData(app.config, app.logger) if app.config.get('INFLUXDB_HOST') else None
     app.prometheus_data_provider = PrometheusData(app.config, app.logger) if app.config.get('PROMETHEUS_HOST') else None
     
-    # Reading all external configs, setting variables to `None` if files are not present
+    # Reading tables configs, setting variable to `None` if file is not present
     tables = get_yaml_file_content(TABLES_PATH)
-    reports = get_yaml_file_content(REPORTS_PATH)
-    extra_etls_file = get_yaml_file_content(EXTRA_ETL_PATH)
-    extra_etls = [ExtraEtl(**etl) for etl in extra_etls_file] if extra_etls_file else None
 
     app.table_data_provider = TableDataProvider(
         app.airflow_data_provider, app.influx_data_provider,
         app.prometheus_data_provider, tables, app.logger, app.config) if tables else None
 
     app.etl_data_provider = EtlDataProvider(
-        app.config, extra_etls, app.airflow_data_provider, app.table_data_provider)
-
-    app.description_data_provider = DescriptionData(
-        app.config, app.logger, tables) if app.config.get('TABLE_DESCRIPTION_ACTIVE') else None
-
-    app.report_data_provider = ReportsDataProvider(
-        app.airflow_data_provider, reports['reports']
-    ) if reports else None
+        app.config, app.airflow_data_provider, app.table_data_provider)
 
     app.async_request_executor = ThreadPoolExecutor(max_workers=3)
+    app.cache = SimpleCache()
```

Comment:
Default cache is 300 seconds, can we bump it to 60 * 60 * 24? The table descriptions change very rarely.

Reply:
Good question! With 1-day TTL for the cache we should provide some mechanism to clear the cache without restarting the app to make the process of documenting tables easier... But I think this is rather "hacky", maybe a better solution would be to make this value configurable via `settings.py`?